src/main/java/uk/gov/ons/ssdc/notifysvc/messaging/ManagedMessageRecoverer.java

```diff
@@ -161,12 +164,35 @@ private void logMessage(
     }
 
     if (logStackTraces) {
+      // Having a separate event for when we are rate limited - makes it easier to track
+      if (cause.getCause() != null
+          && cause.getCause().getMessage() != null
+          && cause.getCause().getMessage().contains(rateLimiterExceptionMessage)) {
+        log.atError()
+            .setMessage("Could not process message - rate limited")
+            .setCause(cause)
+            .addKeyValue("message_hash", messageHash)
+            .log();
+        return;
```

Comment:
It feels a bit odd doing `if ... return` rather than just nesting an `if ... else` here